estate/models/estate_property.py

```diff
@@ -0,0 +1,97 @@
+        help="Defines the direction in which the garden in oriented")
+
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        string="Status",
+        selection=[("new", "New"), ("offer received", "Offer Received"), ("offer accepted", "Offer Accepted"), ("sold", "Sold"), ("canceled", "Canceled")],
+        default="new",
+        help="Choose the state of the Offer")
+
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    buyer_id = fields.Many2one("res.partner", string="Buyer")
+    salesperson_id = fields.Many2one('res.users', string='Salesman', default=lambda self: self.env.user)
+
+    tag_ids = fields.Many2many("estate.property.tag")
+
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+
+    total_area = fields.Float(compute="_compute_total_area")
+
+    best_price = fields.Float(compute="_calculate_best_price", string="Best Offer")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+
+    @api.depends("offer_ids")
+    def _calculate_best_price(self):
+        self.best_price = 0
+        for offers in self.offer_ids:
+            if self.best_price < offers.price:
+               self.best_price = offers.price
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        for property in self:
+            if property.garden:
+                property.garden_area = 10
+                property.garden_orientation = "north"
+            else:
+                property.garden_area = 0
+                property.garden_orientation = ""
+
+    def set_estate_property_to_sold(self):
+        for property in self:
+            if property.state == "canceled":
+                raise UserError("Canceled properties cannot be sold!")
+            else:
+                property.state = "sold"
+        return True
+
+    def set_estate_property_to_canceled(self):
+        for property in self:
+            if property.state == "sold":
+                raise UserError("Sold properties cannot be canceled!")
+            else:
+                property.state = "canceled"
+        return True
+
+    @api.constrains('selling_price','expected_price')
+    def _check_selling_price(self):
+        for record in self:
+            if not float_is_zero(record.selling_price, precision_digits=2):
+                if float_compare(record.selling_price, record.expected_price * 0.9, precision_digits=2) == -1:
+                    raise ValidationError("Selling Price cannot be lower than 90% of the Expected Price!")
+
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)', 'The selling price must be strictly positive!'),
+        ('check_selling_price', 'CHECK(selling_price >= 0)', 'The expected price must be positive!')
+    ]
```

Comment:
please put all the private attributes together => _sql constraints right after name, description and order (for good practice, we expect it to be there)